File: java/parser/src/main/java/gov/lbl/parser/parser/Class_modificationVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Argument_list;
import gov.lbl.parser.domain.Class_modification;

public class Class_modificationVisitor extends modelicaBaseVisitor<Class_modification> {
    @Override
    public Class_modification visitClass_modification(modelicaParser.Class_modificationContext ctx) {
        Argument_listVisitor argument_listVisitor = new Argument_listVisitor();
        Argument_list argument_list = ctx.argument_list().accept(argument_listVisitor);
        
        return new Class_modification(argument_list);
    }   
}

File: jsParser/parser/Class_modificationVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Class_modification = require('../domain/Class_modification')

const Argument_listVisitor = require('./Argument_listVisitor')

class Class_modificationVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitClass_modification (ctx) {
    let argument_list = null

    if (ctx.argument_list()) {
      const argument_listVisitor = new Argument_listVisitor.Argument_listVisitor()
      argument_list = argument_listVisitor.visitArgument_list(ctx.argument_list())
    }

    return new Class_modification.Class_modification(argument_list)
  }
};

Class_modificationVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitClass_modification = this.visitClass_modification
exports.Class_modificationVisitor = Class_modificationVisitor
